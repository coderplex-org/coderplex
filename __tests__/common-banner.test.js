import React from 'react';
import { shallow } from 'enzyme';

import CommonBanner from '../components/common-banner';

describe('Testing CommonBanner of `components/common-banner`', () => {
  const shallowWrapper = shallow(<CommonBanner />);

  it('Check the snapshot', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('should have title tag rendered', () => {
    expect(shallowWrapper.find('h1').length).toBe(1);
  });

  it('should have subtitle tag rendered', () => {
    expect(shallowWrapper.find('h2').length).toBe(1);
  });

  describe('should render the props', () => {
    const pageTitle = 'title of the page';
    const pageSubTitle = 'Subtitle of the page';
    const rootWrapper = shallow(
      <CommonBanner pageTitle={pageTitle} pageSubTitle={pageSubTitle} />,
    );

    it('should display title', () => {
      const headerElement = rootWrapper.find('.headline');
      expect(headerElement.props().children).toEqual(pageTitle);
    });

    it('should display subtitle', () => {
      const subHeaderElement = rootWrapper.find('h2');
      expect(subHeaderElement.props().children).toEqual(pageSubTitle);
    });
  });
});
