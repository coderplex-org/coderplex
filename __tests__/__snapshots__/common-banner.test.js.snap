// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CommonBanner of \`components/common-banner\` Check the snapshot 1`] = `
<div
  className="jsx-1924801089"
>
  <div
    className="jsx-1924801089 root"
  >
    <h1
      className="jsx-1924801089 headline"
    />
    <h2
      className="jsx-1924801089"
    />
  </div>
  <JSXStyle
    css=".root.jsx-1924801089{background-color:#f4f7fb;min-height:150px;text-align:center;padding-top:30px;padding-bottom:30px;}.headline.jsx-1924801089{font-size:4em;color:#df1cb5;font-weight:900;}h2.jsx-1924801089{max-width:1024px;margin-left:auto;margin-right:auto;-webkit-letter-spacing:2px;-moz-letter-spacing:2px;-ms-letter-spacing:2px;letter-spacing:2px;line-height:2;}@media (max-width:720px){h2.jsx-1924801089{font-size:14px;padding:0 10px;}}"
    styleId="1924801089"
  />
</div>
`;
